transformers/shared/transformer_utils.go

```diff
@@ -52,3 +54,19 @@ func VerifyLog(log types.Log, expectedNumTopics int, isDataRequired bool) error
 	}
 	return nil
 }
+
+func GetLogNoteAddresses(arrayLength uint64, eventLogData []byte) ([]string, error) {
+	startingIndex := 2
+	maxVisibleAddresses := 4
+	endIndex := startingIndex + int(math.Min(float64(maxVisibleAddresses), float64(arrayLength))-1)
+	var addresses []string
+	for i := startingIndex; i <= endIndex; i++ {
+		logDataBytes, logNoteErr := GetLogNoteArgumentAtIndex(i, eventLogData)
+		if logNoteErr != nil {
+			return nil, logNoteErr
+		}
+		addressHex := common.BytesToAddress(logDataBytes).Hex()
+		addresses = append(addresses, addressHex)
```

Comment:
Worth noting that we're never actually adding the addresses in this list to `public.addresses`. We could hypothetically return an array of address FKs rather than the addresses themselves, but Postgres can't enforce referential integrity on an array of FKs, so I thought it would be safer to just use the raw addresses.

Maybe it would make sense to add a `GetOrCreateAdress` line in here, just so we have it in the `addresses` table, even if we aren't referencing it? But I figured that would be happening in the storage transformer anyway, so there's not much need to do it here. Interested to hear people's thoughts.

Reply:
Good call-- definitely worth thinking about, but so far I'm not really finding that it's any tougher when the addresses aren't in `public.addresses`, since we pretty much always interact with that table using the `GetOrCreateAddress` function.

And on the `storageRepository` side of things, we ultimately need a list of addresses (not address IDs) for the key loader, so it's pretty easy to just query them directly from the array in `median_kiss_batch`/`median_diss_batch` without doing any joins or anything.